### PR TITLE
Field-level error states for scheduled payment form

### PR DESCRIPTION
**Pull request by lukemelia**
- Revealed once the user has interacted with the field

The border colors were not captured that well by this video, but you can get the basic idea, and try it out locally.

https://user-images.githubusercontent.com/353/205231573-61eede99-8447-4516-a60d-efe273692117.mov

### CODE REVIEW

**Review by jurgenwerk — APPROVED** [review]
I appreciate the hand written validator. I always seem to struggle with 3rd party form validation libs.

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/validator.ts [hunk 1]
```diff
@@ -39,37 +33,88 @@ export default class SchedulePaymentFormValidator {
...
+  get paymentTypeErrorMessage(): string {
     const { form } = this;
     if (form.selectedPaymentType === 'one-time') {
-      return !!form.paymentDate;
+      if (!form.paymentDate) {
+        return 'must choose a payment date and time';
+      }
+    } else if (form.selectedPaymentType === 'monthly') {
+      if (!form.paymentDayOfMonth) {
+        return 'must choose an payment day of month';
```
Comment (backspace) [inline packages/safe-tools-client/app/components/schedule-payment-form-action-card/validator.ts:55]:
```suggestion
        return 'must choose a payment day of month';
```

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/validator.ts [hunk 2]
```diff
@@ -39,37 +33,88 @@ export default class SchedulePaymentFormValidator {
   get isValid() {
     return (
       this.isPaymentTypeValid &&
-      this.isRecipientValid &&
+      this.isRecipientAddressValid &&
       this.isAmountValid &&
       this.isGasTokenValid &&
       this.isMaxGasFeeValid
```
Comment (backspace) [inline packages/safe-tools-client/app/components/schedule-payment-form-action-card/validator.ts:39]: Thoughts re validation of dates? Currently I can choose a one-time payment in the past or an until in the past for recurring.

**Review by backspace — APPROVED** [review]

**Comment by backspace** [comment]
This predates your PR as I see it [on `main`](https://safe-tools-staging.stack.cards/schedule) but I noticed this while trying it out, that after switching to recurring I can’t switch back to one-time:

cannot-switch-back-to-one-time

There are no console errors 🤔

File: packages/safe-tools-client/app/components/schedule-payment-form-action-card/validator.ts (hunk 2 above)
Comment (lukemelia) [reply packages/safe-tools-client/app/components/schedule-payment-form-action-card/validator.ts:39]: Yeah, we should add validation for this, or a minimum DateTime to the picker.

Comment (lukemelia) [reply packages/safe-tools-client/app/components/schedule-payment-form-action-card/validator.ts:39]: captured this in Linear

**Comment by lukemelia** [comment]
>This predates your PR as I see it [on main](https://safe-tools-staging.stack.cards/schedule) but I noticed this while trying it out, that after switching to recurring I can’t switch back to one-time

Thanks. I captured this in Linear.

**Pull request merged**